fifo_queue.go

```diff
@@ -219,3 +219,29 @@ func (st *FIFO) IsLocked() bool {
 
 	return st.isLocked
 }
+
+// Swap swaps values from position a to position b and vice versa.
+func (st *FIFO) Swap(a int, b int) *QueueError {
+	if st.isLocked {
+		return NewQueueError(QueueErrorCodeLockedQueue, "The queue is locked")
+	}
+
+	st.rwmutex.Lock()
+	defer st.rwmutex.Unlock()
+
+	length := len(st.slice)
+	if length == 0 {
+		return NewQueueError(QueueErrorCodeEmptyQueue, "Empty queue")
+	}
+
+	if a == b {
+		return NewQueueError(QueueErrorCodeIndexesMatch, "Indexes are the same number")
+	}
+
+	if a > length || b > length {
```

Comment:
This should be `>=` because arrays are 0-based.

As an example, assume you have an array of length `3` with elements `[0, 1, 2]`, calling `Swap(1, 3)` will not raise an error because `3 > length` is `false`.

Please add two tests for this operation: one using `length-1` as one of the positions and another using `length` as one of the positions. The first one should succeed and the second one should fail.